src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,87 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed. Annotation @Component is missed in "
+                    + clazz.getName());
+        }
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field: declaredFields) {
+            clazzImplementationInstance =
+                    getClazzImplementationInstance(clazzImplementationInstance, clazz, field);
+        }
+        return clazzImplementationInstance != null
+                ? clazzImplementationInstance
+                : createNewInstance(clazz);
+    }
+
+    private Object getClazzImplementationInstance(
+            Object clazzImplementationInstance, Class<?> clazz, Field field) {
+        if (field.isAnnotationPresent(Inject.class)) {
+            Object fieldInstance = getInstance(field.getType());
+            clazzImplementationInstance = createNewInstance(clazz);
+            try {
+                field.setAccessible(true);
+                field.set(clazzImplementationInstance, fieldInstance);
+            } catch (IllegalAccessException e) {
+                throw new RuntimeException("Can`t initialize field "
+                        + field.getName()
+                        + "Class: "
+                        + clazz.getName()
+                        + e);
+            }
+        }
```

Comment:
i suggest adding the following code before return statement
        if (clazzImplementationInstance == null) {
            clazzImplementationInstance = createNewInstance(clazz);
        }